common/chat.cpp

```diff
@@ -449,13 +449,7 @@ std::string common_chat_format_name(common_chat_format format) {
     }
 }
 
-const common_grammar_options grammar_options {
-    /* .dotall = */ false,
-    /* .compact_spaces = */ false,
-    // /* .compact_spaces = */ true,
-};
-
-static bool parse_json(std::string::const_iterator & it, const std::string::const_iterator & end, json & out) {
+static std::optional<json> parse_json(std::string::const_iterator & it, const std::string::const_iterator & end) {
```

Comment:
We could also remove the `std::optional` here:

```suggestion
static bool parse_json(std::string::const_iterator & it, const std::string::const_iterator & end, json & output) {
```

Then use it:

```cpp
json output;
bool ok = parse_json(..., output);
if (ok) {
  ...
}
```

Reply:
applied, thanks